src/test/rehydration.test.js

```diff
@@ -421,8 +435,33 @@ describe('rehydration', () => {
       expectCSSMatches(`
         @-webkit-keyframes keyframe_880 {from {opacity: 0;}}@keyframes keyframe_880 {from {opacity: 0;}}
         .sc-a { } .d { -webkit-animation:keyframe_880 1s both; animation:keyframe_880 1s both; }
+        .sc-b { } .c { -webkit-animation:keyframe_144 1s both; animation:keyframe_144 1s both; }
         @-webkit-keyframes keyframe_144 {from {opacity:1;}}@keyframes keyframe_144 {from {opacity:1;}}
+      `)
+    })
+
+    it('should pass the keyframes name through props along as well', () => {
+      const fadeIn = keyframes`
+        from { opacity: 0; }
+      `
+      const A = styled.div`
+        animation: ${props => props.animation} 1s both;
+      `
+      const fadeOut = keyframes`
+        from { opacity: 1; }
+      `
+      const B = styled.div`
+        animation: ${props => props.animation} 1s both;
+      `
+      /* Purposely rendering out of order to make sure the output looks right */
+      mount(<B animation={fadeOut} />)
+      mount(<A animation={fadeIn} />)
+
+      expectCSSMatches(`
+        @-webkit-keyframes keyframe_880 {from {opacity: 0;}}@keyframes keyframe_880 {from {opacity: 0;}}
+        .sc-a { } .d { -webkit-animation:keyframe_880 1s both; animation:keyframe_880 1s both; }
         .sc-b { } .c { -webkit-animation:keyframe_144 1s both; animation:keyframe_144 1s both; }
+        @-webkit-keyframes keyframe_144 {from {opacity:1;}}@keyframes keyframe_144 {from {opacity:1;}}
```

Comment:
Is it a problem that this is injected after the usage?

Reply:
@probablyup Look at example 3 https://www.w3.org/TR/css-animations-1/#at-ruledef-keyframes

It's ok to define keyframes elsewhere from it usage.